platform/state/platform_state.proto

```diff
@@ -0,0 +1,391 @@
+ * This message records the minimum ancient threshold agreed by all judges in
+ * a round.
+ */
+message MinimumJudgeInfo {
+    /**
+     * A consensus round.<br/>
+     * The round this judge information applies to.
+     */
+    uint64 round = 1;
+
+    /**
+     * Minimum ancient threshold for all judges in the round.
+     * <p>
+     * This SHALL reflect the relevant minimum threshold, whether
+     * generation-based or birth-round-based.
+     */
+    uint64 minimum_judge_ancient_threshold = 2;
+}
+
+/**
+ * A network address book.<br/>
+ * The address book records the address of every known consensus node that
+ * participates in the network, including `0 weight` nodes.<br/>
+ */
+message AddressBook {
+    /**
+     * A consensus round.<br/>
+     * The round when this address book was created.
+     */
+    int64 round = 1;
+
+    /**
+     * The node ID of the next address that can be added.
+     * <p>
+     * The next node identifier assigned SHALL be equal to this value.<br/>
+     * All existing node identifiers SHALL be strictly less than this value.<br/>
+     */
+    NodeId next_node_id = 2;
+
+    /**
+     * A list of all consensus node addresses.
+     * <p>
+     * This list SHALL NOT be empty.
+     * If a consensus node is not in this list it SHALL NOT participate in the
+     * network.
+     */
+    repeated Address addresses = 3;
+}
+
+/**
+ * A single network address.
+ * This is one address in the network address book, including all required
+ * information to include that consensus node in the consensus gossip.
+ *
+ * ### Implementation Detail
+ * When `enableEventStreaming` is set to true, the memo field is REQUIRED
+ * and MUST be unique.
+ */
+message Address {
+    /**
+     * The ID of this member.
+     * <p>
+     * This identifier SHALL be agreed upon by all consensus nodes.
+     */
+    NodeId id = 1;
+
+    /**
+     * The nickname this consensus node uses to refer to another consensus node.
+     */
+    string nickname = 2;
+
+    /**
+     * The name that a consensus node uses to refer to themselves.
+     */
+    string self_name = 3;
+
+    /**
+     * A consensus weight.<br/>
+     * This value is the relative weight for this consensus node in the
+     * consensus voting algorithm.
+     */
+    uint64 weight = 4;
+
+    /**
+     * A string network address.<br/>
+     * This value is the hostname assigned on the "internal" network
+     * interface behind any network firewalls, protocol gateways, or
+     * DNS translations.
+     * <p>
+     * This value SHALL be either a FQDN host name or an IPv4 address.
+     */
+    string hostname_internal = 5;
+
+    /**
+     * Network port number.<br/>
+     * This value is the "translated" port number assigned on the "internal"
+     * network behind any network firewalls or other address translations.
+     * <p>
+     * This value SHALL be between 1 and 65535.
+     */
+    uint32 port_internal = 6;
```

Comment:
Before committing to supporting these fields forever, it might be worth having another discussion about this. I never quite wrapped my head around the truly irreplaceable need for these to be present in the address book.

Reply:
Who should I have this conversation with?

Reply:
I think devops are going to be the important people to talk to on this one. They should be knowledgeable about whether or not this is in any way relevant. @nathanklick should also probably be included in the conversation.